x-pack/test/detection_engine_api_integration/utils/delete_all_signals.ts

```diff
@@ -7,25 +7,37 @@
 
 import type SuperTest from 'supertest';
 import type { ToolingLog } from '@kbn/tooling-log';
+import type { Client } from '@elastic/elasticsearch';
 import { DETECTION_ENGINE_INDEX_URL } from '@kbn/security-solution-plugin/common/constants';
 import { countDownTest } from './count_down_test';
 
 /**
- * Deletes the signals index for use inside of afterEach blocks of tests
- * @param supertest The supertest client library
+ * Deletes all alerts from a given index or indices, defaults to `.alerts-security.alerts-default`
+ * For use inside of afterEach blocks of tests
  */
-export const deleteSignalsIndex = async (
+export const deleteAllSignals = async (
   supertest: SuperTest.SuperTest<SuperTest.Test>,
-  log: ToolingLog
+  log: ToolingLog,
+  es: Client,
+  index: string[] = ['.alerts-security.alerts-default']
 ): Promise<void> => {
   await countDownTest(
     async () => {
       await supertest.delete(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send();
```

Comment:
Is this call (and the endpoint itself) doing something meaningful at this point? If there's no value to this call, we should remove it.

Taking that a step further to the endpoint itself, we should either a) update the documentation to reflect the current  (and historic) purpose of this endpoint, and/or b) deprecate the endpoint.

cc @ymao1

Reply:
There were tests that weren't passing without it, here's one of the [failing buildkite jobs](https://buildkite.com/elastic/kibana-pull-request/builds/123734#0187c013-aa7b-473e-b308-fe64e8718bf0). All of them had basically the same error message but there were only 4(?) that were failing so perhaps the problem is more with those tests?

Reply:
Worst case, we could probably factor out that call since I think for 95% of the tests, we don't need it and won't need it for any going forward

Reply:
Those failures are interesting; I would agree that it's not clear whether that's a valid state a user could get themselves into, or an issue with the tests.

There's definitely still _something_ that that DELETE route of ours does, but I think the broader point is that the system has changed and implicitly so has that route's role (as evidenced by this PR). We should update documentation to reflect this change. cc @jmikell821

Reply:
@rylnd can you create an issue in the `security-docs` repo with what docs we need to update so we have more context? Thanks!

Reply:
@jmikell821 absolutely! I should have pinged the docs team as an FYI, not you individually, sorry!